[FIX] account: allow users to change products again from catalog

nash-odoo opened the pull request:
**Before this PR:**
When more than 1 line item is added, users are unable to update or remove products from the catalog view.

**Technical Reason:**
There is a function _is_readonly() that returns True if there are multiple line items, and because of this, the value of props.readOnly becomes True which prevents users from updating the products in catalog view.

**Solution:**
Adding one more condition in account_move_line to satisfy the desired behavior.

**After this PR:**
The users can update or remove the products from the catalog view if there are multiple line items.

**Task**-3806509

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/157994).

nash-odoo commented:
Hello @epictete,
any update on this PR? Thanks :blush:

epictete commented:
Hello @nash-odoo 
I'm not an expert in Sales / Products, but it seems that this condition was added on purpose: https://github.com/odoo/odoo/pull/146262#discussion_r1446214937

nash-odoo commented:
Hello @epictete,
Yes, you are right that the condition was written on purpose, but with that fix, the condition is not working properly. It doesn't check if the same product is repeated or not, it just checks if there are multiple line_ids then it is True.

Now, I removed the condition from there and added another condition like sale_order in account_move
And now, with this fix, only products that repeat in multiple line_ids will become readOnly.

Thanks :blush:

epictete reviewed (approved):
@robodoo r+
